interfaces/tests/file_int/file_int.c

```diff
@@ -0,0 +1,559 @@
+/*Tests_SRS_FILE_43_006: [file_destroy shall wait for all pending I / O operations to complete.]*/
+/*Tests_SRS_FILE_43_007: [file_destroy shall close the file handle handle.]*/
+/*Tests_SRS_FILE_43_014: [file_write_async shall enqueue a write request to write source's content to the position offset in the file. ]*/
+/*Tests_SRS_FILE_43_041: If position + size is greater than the size of the file and the call to write is successfull, file_write_async shall grow the file to accomodate the write.]*/
+/*Tests_SRS_FILE_43_008: [file_write_async shall call user_call_back passing user_context and success depending on the success of the asynchronous write operation.]*/
+/*Tests_SRS_FILE_43_030: [file_write_async shall succeed and return FILE_WRITE_ASYNC_OK.]*/
+/*Tests_SRS_FILE_43_021: [file_read_async shall enqueue a read request to read handle's content at position offset and write it to destination. ]*/
+/*Tests_SRS_FILE_43_016: [file_read_async shall call user_callback passing user_context and success depending on the success of the asynchronous read operation.]*/
+/*Tests_SRS_FILE_43_031: [file_read_async shall succeed and return FILE_READ_ASYNC_OK.]*/
+TEST_FUNCTION(large_simultaneous_writes_succeed)
+{
+    ///arrange
+    int block_size = 4096;
+    int num_blocks = 50;
+    WRITE_COMPLETE_CONTEXT contexts[50];
+    unsigned char* sources[50];
+
+    char filename[] = "large_simultaneous_writes_succeed.txt";
+    ASSERT_IS_FALSE(check_file_exists(filename));
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    FILE_HANDLE file_handle = file_create(execution_engine, filename, NULL, NULL);
+
+    for (int i = 0; i < num_blocks; ++i)
+    {
+        sources[i] = (unsigned char*)malloc(block_size);
+        ASSERT_IS_NOT_NULL(sources[i]);
+        memset(sources[i], 'a' + i, block_size);
+        contexts[i].pre_callback_value = num_blocks + 1;
+        interlocked_exchange(&contexts[i].value, contexts[i].pre_callback_value);
+        contexts[i].post_callback_value = i;
+
+        FILE_WRITE_ASYNC_RESULT write_result = file_write_async(file_handle, sources[i], block_size, block_size * i, write_callback, &contexts[i]);
+        ASSERT_ARE_EQUAL(FILE_WRITE_ASYNC_RESULT, FILE_WRITE_ASYNC_OK, write_result);
+    }
+
+    for (int i = 0; i < num_blocks; ++i)
+    {
+        wait_on_address_helper(&contexts[i].value, &contexts[i].pre_callback_value, UINT32_MAX);
+        ASSERT_ARE_EQUAL(int32_t, contexts[i].post_callback_value, interlocked_or(&contexts[i].value, 0), "value should be post_callback_value");
+        ASSERT_IS_TRUE(contexts[i].did_write_succeed);
+    }
+
+    unsigned char* destination = (unsigned char*)malloc(block_size * num_blocks);
+    ASSERT_IS_NOT_NULL(destination);
+    READ_COMPLETE_CONTEXT read_context;
+    read_context.pre_callback_value = 0;
+    interlocked_exchange(&read_context.value, read_context.pre_callback_value);
+    read_context.post_callback_value = 1;
+
+    ///act

```

Comment:
>///act [](start = 4, length = 6)

This seems to be part of assert since you are testing that "large simultaneous writes succeed"